### PR TITLE
Add intial translation of the figures for the chapters of hashing, tree, and heap

### PR DESCRIPTION
**Pull request by krahets**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

### CODE REVIEW

**Comment by krahets** [comment]
Please review the translation based on the following aspects:

Ensure the expression is authentic and accurate.
Maintain consistency between the Chinese and English versions. (Balancing with point 1)
In addition to content, please pay attention to the format:

It is expected to use Roboto font for normal text and Roboto Mono for formulas and code. These may be prone to mistakes since they are set manually.
The style and color of shapes and texts should be consistent between the two versions. (I made some color adjustments in the EN version to enhance readability, so please only report those that feel uncomfortable to you.)

File: en/docs/chapter_hashing/hash_collision.assets/hash_table_chaining.png
Comment (K3v123) [inline en/docs/chapter_hashing/hash_collision.assets/hash_table_chaining.png:1]: the word "address" has been left out on in.
maybe do "chained addresses"?

File: en/docs/chapter_hashing/hash_map.assets/hash_table_reshash.png
Comment (K3v123) [inline en/docs/chapter_hashing/hash_map.assets/hash_table_reshash.png:1]: "capacity doubled" is not in bold unlike in CN it is

File: en/docs/chapter_heap/build_heap.assets/heapify_operations_count.png
Comment (K3v123) [inline en/docs/chapter_heap/build_heap.assets/heapify_operations_count.png:1]: the node looks bold in CN but im not too sure.

File: en/docs/chapter_heap/top_k.assets/top_k_traversal.png
Comment (K3v123) [inline en/docs/chapter_heap/top_k.assets/top_k_traversal.png:1]: the bold was on 1st and 2nd but in en its on the "search" part

File: en/docs/chapter_tree/array_representation_of_tree.assets/array_representation_complete_binary_tree.png
Comment (K3v123) [inline en/docs/chapter_tree/array_representation_of_tree.assets/array_representation_complete_binary_tree.png:1]: no bold text or "complete binary tree"

File: en/docs/chapter_tree/avl_tree.assets/avltree_degradation_from_inserting_node.png
Comment (K3v123) [inline en/docs/chapter_tree/avl_tree.assets/avltree_degradation_from_inserting_node.png:1]: no bold at the bottom eg "balanced binary tree" "unbalanced binary tree" etc

File: en/docs/chapter_tree/avl_tree.assets/avltree_degradation_from_removing_node.png
Comment (K3v123) [inline en/docs/chapter_tree/avl_tree.assets/avltree_degradation_from_removing_node.png:1]: same

File: en/docs/chapter_tree/avl_tree.assets/avltree_left_right_rotate.png
Comment (K3v123) [inline en/docs/chapter_tree/avl_tree.assets/avltree_left_right_rotate.png:1]: "right" and "left" are not in bold

File: en/docs/chapter_tree/binary_tree.assets/balanced_binary_tree.png
Comment (K3v123) [inline en/docs/chapter_tree/binary_tree.assets/balanced_binary_tree.png:1]: same not bold

File: en/docs/chapter_tree/binary_search_tree.assets/bst_remove_case3_step2.png
Comment (K3v123) [inline en/docs/chapter_tree/binary_search_tree.assets/bst_remove_case3_step2.png:1]: sane here and below

File: en/docs/chapter_tree/binary_search_tree.assets/bst_remove_case3_step1.png
Comment (K3v123) [inline en/docs/chapter_tree/binary_search_tree.assets/bst_remove_case3_step1.png:1]: same not bold

File: en/docs/chapter_tree/binary_search_tree.assets/bst_insert.png
Comment (K3v123) [inline en/docs/chapter_tree/binary_search_tree.assets/bst_insert.png:1]: same top is not bold. same as below

File: en/docs/chapter_tree/binary_search_tree.assets/bst_degradation.png
Comment (K3v123) [inline en/docs/chapter_tree/binary_search_tree.assets/bst_degradation.png:1]: same not bold

File: en/docs/chapter_tree/avl_tree.assets/avltree_rotation_cases.png
Comment (K3v123) [inline en/docs/chapter_tree/avl_tree.assets/avltree_rotation_cases.png:1]: not bold

File: en/docs/chapter_tree/avl_tree.assets/avltree_right_rotate_step4.png
Comment (K3v123) [inline en/docs/chapter_tree/avl_tree.assets/avltree_right_rotate_step4.png:1]: same the bold for right rotation

File: en/docs/chapter_tree/avl_tree.assets/avltree_right_rotate_step3.png
Comment (K3v123) [inline en/docs/chapter_tree/avl_tree.assets/avltree_right_rotate_step3.png:1]: right roatation is not in bold same goes for "rotate right" or is it just the style of text?

**Review by K3v123 — COMMENTED** [review]
overall the translations are accurate
its just the bold and colors again.

**Comment by krahets** [comment]
Thanks for your thorough review! @K3v123, @pengchzn  most of the comments have been addressed. A few comments remain unresolved for further discussion. Please let me know if there's anything else that needs fixing.

**Pull request merged**